src/librustc/middle/ty/structural_impls.rs

```diff
@@ -751,6 +658,18 @@ impl<'tcx> TypeFoldable<'tcx> for ty::Predicate<'tcx> {
                 ty::Predicate::ObjectSafe(trait_def_id),
         }
     }
```

Comment:
Seems like maybe we should have `fold_subitems_with` be `unimplemented!` on such cases.